### ISSUE TITLE
Beta-3.0 inconsistent on tcp

### ISSUE THREAD

**Issue by thomasinaz41213** [Lucifer06/RemoteGPIO#17]
This is just an FYI. I don't want you to waste any time looking into this until I have had a chance to do some troubleshooting.

As the title says, I am having trouble with the beta version operating on tcp. I have managed to get it to work once but not since. I will do some more tinkering and see if I can narrow anything down. I have been reverting back to my own fork of the v2.xx because I have 3 modules on tcp. When I have a little more time to dive into it I will try a fresh firmware and see if that makes a difference. Sometimes things get left over when switching back and forth between packages that do similar things. I will also try putting the modules on the router instead of thru the Venus AP. I wouldn't expect that to make a difference but I would rather eliminate as many factors as possible. I will report back with any new findings.

**Comment by Lucifer06** [comment]
Thomas, don't forget to set correctly the Modbus ID, even with TCP. So for unit2, the Modbus ID needs to be set to 2. Cf screenshot.
And TCP protocol must be set to Modbus-RTU over TCP.
![Capture d’écran 2024-03-11 à 20 11 19](https://github.com/Lucifer06/RemoteGPIO/assets/10178879/41c185d9-7a09-4b1d-9d83-ef5db7c50763)

This is working like this on my side.
And just in case I'm also showing you HW and SW version.
[
![Capture d’écran 2024-03-11 à 20 12 48](https://github.com/Lucifer06/RemoteGPIO/assets/10178879/9cb84006-7f8f-4306-88d1-57d5b3779ca7)
](url)

**Comment by thomasinaz41213** [comment]
Turns out to be a permission issue. For some reason the rgpio_driver is not being installed with execute permission. If I ran the driver from console the relays would work but not otherwise. I checked the driver log and found permission denied. Chmod 755 and reboot and all is well. I have another relay module on the way so I can have one connected to my test system. Then I don't have to test things on my production system. I'll put this on the shelf for now.
